### PR TITLE
modify all api response to the same format (JSONResponse with status_code and content set)

### PR DESCRIPTION
**Pull request by assertion**
Thanks for @Umpire2018 's suggestion, cc @rbren , modify the response format to the same like 

```
# success:
return JSONResponse(
        status_code=status.HTTP_200_OK,
        content=models,
    )
# failed:
return JSONResponse(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            content={'error': error_msg},
        )
```

previous pr: https://github.com/OpenDevin/OpenDevin/pull/1158

### CODE REVIEW

File: opendevin/server/listen.py [hunk 1]
```diff
@@ -134,8 +150,14 @@ def select_file(file: str):
...
         logger.error(f'Error opening file {file}: {e}', exc_info=False)
         error_msg = f'Error opening file: {e}'
-        return Response(f'{{"error": "{error_msg}"}}', status_code=500)
-    return {'code': content}
+        return JSONResponse(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            content={'error': error_msg},
+        )
+    return JSONResponse(
+        status_code=status.HTTP_200_OK,
```
Comment (rbren) [inline opendevin/server/listen.py:158]: I'm going to guess that `200` is the default, so maybe we don't need to specify these?

Comment (Umpire2018) [reply opendevin/server/listen.py:158]: It's true that many web frameworks default to a `200` OK status for successful HTTP responses, explicitly setting the status code can have benefits for code consistency and clarity

**Review by Umpire2018 — APPROVED** [review]

File: opendevin/server/listen.py (hunk 1 above)
Comment (li-boxuan) [reply opendevin/server/listen.py:158]: Being more concise seems more pythonic to me.

Comment (assertion) [reply opendevin/server/listen.py:158]: So maybe for `200` status code response, just return the `content` without `JSONResponse()`  and `status_code` specified. Like this:

```
@app.get('/api/refresh-files')
def refresh_files():
    structure = files.get_folder_structure(Path(str(config.get('WORKSPACE_BASE'))))
    return structure.to_dict()


@app.get('/api/select-file')
def select_file(file: str):
    try:
        # xxx
    except Exception as e:
        # yyy
        return JSONResponse(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            content={'error': error_msg},
        )
    return {'code': content}


```

Comment (rbren) [reply opendevin/server/listen.py]: 👍 I like that

**Pull request merged**